zcash_primitives/src/transaction/components/amount.rs

```diff
@@ -281,6 +281,19 @@ impl NonNegativeAmount {
         let amount = u64::from_le_bytes(bytes);
         Self::from_u64(amount)
     }
+
+    /// Reads a NonNegativeAmount from a signed 64-bit little-endian integer.
```

Comment:
An `i64` is numerically a signed 63-bit integer, not a signed 64-bit integer. This gets confusing when talking about currency, so we need to be clear here. We might want to replicate or adapt the warning I wrote for the `orchard::value` module (which I copied into `sapling::value` as well IIRC): https://github.com/zcash/orchard/blob/c48247dd34c53af43697e97837c81a5b37ad98c8/src/value.rs#L23-L31